### PR TITLE
Lg/benchmark cmd reference

### PR DESCRIPTION
**Pull request by lisa-parity**

### CODE REVIEW

**Comment by lisa-parity** [comment]
@ggwpez , if you get a chance, please take a look at the command reference section I threw together 😄

File: content/md/en/docs/reference/command-line-tools/node-template.md [hunk 1]
```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
...
+
+```shell
+./target/release/node-template benchmark pallet \
+    --chain dev \
+    --extrinsic do_something \
+    --pallet pallet_template \
+    --steps 10 \
+    --repeat 20 \
+    --output pallets/weight2.rs
+```
```
Comment (ggwpez) [inline content/md/en/docs/reference/command-line-tools/node-template.md:251]: You probably already did it in another spot; but I think can add some output of how the stuff should look like.  
The `--list` from above produces:  
```pre
pallet, benchmark
frame_benchmarking, addition
frame_benchmarking, subtraction
...
pallet_timestamp, set
pallet_timestamp, on_finalize
pallet_template, do_something
```

and this one:
<details>
  <summary>spoiler</summary>

```pre
Pallet: "pallet_template", Extrinsic: "do_something", Lowest values: [], Highest values: [], Steps: 10, Repeat: 20
Raw Storage Info
========
Storage: TemplateModule Something (r:0 w:1)

Median Slopes Analysis
========
-- Extrinsic Time --

Model:
Time ~=     8.08
    + s        0
              µs

Reads = 0 + (0 * s)
Writes = 1 + (0 * s)

Min Squares Analysis
========
-- Extrinsic Time --

Data points distribution:
    s   mean µs  sigma µs       %
    0     11.73     0.942    8.0%
   11     7.914     0.077    0.9%
   22     7.918     0.095    1.1%
   33     7.989      0.08    1.0%
   44     8.034     0.095    1.1%
   55     7.918      0.07    0.8%
   66     8.198      0.16    1.9%
   77     8.052     0.088    1.0%
   88     8.078     0.065    0.8%
  100     7.976     0.135    1.6%

Quality and confidence:
param     error
s         0.003

Model:
Time ~=    9.235
    + s        0
              µs

Reads = 0 + (0 * s)
Writes = 1 + (0 * s)
```

</details>

File: content/md/en/docs/reference/command-line-tools/node-template.md [hunk 2]
```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
...
+
+To measure the average, median, minimum, and maximum execution time per-block and per-extrinsic, you can run the `node-template benchmark overhead` subcommand:
+
+```shell
+./target/release/node-template benchmark overhead
+```
```
Comment (ggwpez) [inline content/md/en/docs/reference/command-line-tools/node-template.md:257]: If you are interested in adding some output here:  
```pre
Running 10 warmups...    
Executing block 100 times    
Per-block execution overhead [ns]:
	Total: 81779519
	Min: 695173, Max: 1522917
	Average: 817795, Median: 763360, Stddev: 160759.2
	Percentiles 99th, 95th, 75th: 1443343, 1263935, 814356    
	Writing weights to "block_weights.rs"    
Running 10 warmups...    
Executing block 100 times    
Building block, this takes some time...    
Extrinsics per block: 12000    
Running 10 warmups...    
Executing block 100 times
Per-extrinsic execution overhead [ns]:
	Total: 8955542
	Min: 85950, Max: 97448
	Average: 89555, Median: 88348, Stddev: 2713.05
	Percentiles 99th, 95th, 75th: 97247, 95040, 91733    
	Writing weights to "extrinsic_weights.rs"
```
Feel free to ignore if if you think if fits better at a different spot.

File: content/md/en/docs/reference/command-line-tools/node-template.md [hunk 3]
```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
...
+
+By default, the command executes the benchmark 100 times, generates results, and writes the output to the `block_weights.rs` and `extrinsics_weights.rs` files.
+You can use command-line options to adjust the calculated weight by adding units or by multiplying the average execution time by some factor.
+
+To measure the storage execution time for the Substrate development chain specification, you can run the following command:
+
+```shell
+./target/release/node-template benchmark storage --state-version 1
 ```
```
Comment (ggwpez) [inline content/md/en/docs/reference/command-line-tools/node-template.md:266]: Same here with the output:  
```pre
Warmup round 1/1    
Preparing keys from block BlockId::Number(0)    
Reading 36 keys    
Time summary [ns]:
	Total: 478367
	Min: 8889, Max: 32258
	Average: 13287, Median: 12580, Stddev: 4189.73
	Percentiles 99th, 95th, 75th: 32258, 21863, 14563
Value size summary:
	Total: 157217
	Min: 1, Max: 155944
	Average: 4367, Median: 16, Stddev: 25621.19
	Percentiles 99th, 95th, 75th: 155944, 82, 80    
Warmup round 1/1    
Preparing keys from block BlockId::Number(0)    
Writing 36 keys    
Time summary [ns]:
	Total: 1877401
	Min: 13476, Max: 665226
	Average: 52150, Median: 25740, Stddev: 111805.24
	Percentiles 99th, 95th, 75th: 665226, 252697, 30427
Value size summary:
	Total: 157217
	Min: 1, Max: 155944
	Average: 4367, Median: 16, Stddev: 25621.19
	Percentiles 99th, 95th, 75th: 155944, 82, 80    
Writing weights to "rocksdb_weights.rs"
```

The two important arguments for this one are `--db paritydb` if you want to use a ParityDB snapshot instead.  
And to use `--state-version 0` for real chain snapshots, as the `--help` indicates. You can also link the [README](https://github.com/paritytech/substrate/tree/master/utils/frame/benchmarking-cli/src/storage#the-benchmark-storage-command) which explains it a bit.

**Review by ggwpez — APPROVED** [review]
Thanks for updating all of this!  
That will mean a lot less people pinging me :laughing:

File: content/md/en/docs/reference/command-line-tools/node-template.md [hunk 4]
```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
...
+./target/release/node-template benchmark pallet \
     --chain dev \
-    --execution wasm \
-    --wasm-execution compiled \
-    --extrinsic '\*' \
-    --pallet '\*' \
-    --output ./pallets/example/weights.rs
+    --extrinsic "*" \
+    --pallet "*" \
+    --output pallets/weights.rs
```
[the remark this answers is not in the document]
Comment (lisa-parity) [reply content/md/en/docs/reference/command-line-tools/node-template.md:235]: Using --list with the simpler command below.

File: content/md/en/docs/reference/command-line-tools/node-template.md (hunk 1 above)
Comment (lisa-parity) [reply content/md/en/docs/reference/command-line-tools/node-template.md:309]: Added --list output and a sample output

**Pull request merged**